srp/tests/rfc5054.rs

```diff
@@ -0,0 +1,148 @@
+use num_bigint::BigUint;
+use sha1::Sha1;
+use srp::client::SrpClient;
+use srp::groups::G_1024;
+use srp::server::SrpServer;
+use srp::utils::{compute_k, compute_u};
+
+fn strip_white(s: &str) -> String {
+    s.replace(" ", "").replace("\n", "")
+}
+
+#[test]
+fn rfc5054() {
+    let i = b"alice";
+    let p = b"password123";
+    let s = hex::decode(strip_white("BEB25379 D1A8581E B5A72767 3A2441EE")).unwrap();
+    let group = &G_1024;
+
+    let k = compute_k::<Sha1>(group);
+
+    assert_eq!(
+        k.to_bytes_be(),
+        hex::decode(strip_white("7556AA04 5AEF2CDD 07ABAF0F 665C3E81 8913186F")).unwrap(),
+        "bad k value"
+    );
+
+    let identity_hash = SrpClient::<Sha1>::compute_identity_hash(i, p);
+    let x = SrpClient::<Sha1>::compute_x(identity_hash.as_slice(), &s);
+
+    assert_eq!(
+        x.to_bytes_be(),
+        hex::decode(strip_white("94B7555A ABE9127C C58CCF49 93DB6CF8 4D16C124")).unwrap(),
+        "bad x value"
+    );
+
+    let client = SrpClient::<Sha1>::new(group);
+    let v = client.compute_v(&x);
+
+    assert_eq!(
+        v.to_bytes_be(),
+        hex::decode(strip_white(
+            "        
+         7E273DE8 696FFC4F 4E337D05 B4B375BE B0DDE156 9E8FA00A 9886D812
+         9BADA1F1 822223CA 1A605B53 0E379BA4 729FDC59 F105B478 7E5186F5
+         C671085A 1447B52A 48CF1970 B4FB6F84 00BBF4CE BFBB1681 52E08AB5
+         EA53D15C 1AFF87B2 B9DA6E04 E058AD51 CC72BFC9 033B564E 26480D78
+         E955A5E2 9E7AB245 DB2BE315 E2099AFB
+         "
+        ))
```

Comment:
With the `hex-literal` crate this becomes:

```suggestion
        hex!(
            "        
         7E273DE8 696FFC4F 4E337D05 B4B375BE B0DDE156 9E8FA00A 9886D812
         9BADA1F1 822223CA 1A605B53 0E379BA4 729FDC59 F105B478 7E5186F5
         C671085A 1447B52A 48CF1970 B4FB6F84 00BBF4CE BFBB1681 52E08AB5
         EA53D15C 1AFF87B2 B9DA6E04 E058AD51 CC72BFC9 033B564E 26480D78
         E955A5E2 9E7AB245 DB2BE315 E2099AFB
         "
        )
```

It automatically handles stripping whitespace.